cmd/spire-server/cli/run/run_test.go

```diff
@@ -73,27 +75,646 @@ func TestParseFlagsGood(t *testing.T) {
+				require.Nil(t, c)
+			},
+		},
+		{
+			msg:         "invalid log_format returns an error",
+			expectError: true,
+			input: func(c *config) {
+				c.Server.LogFormat = "not-a-valid-format"
+			},
+			test: func(t *testing.T, c *server.Config) {
+				require.Nil(t, c)
+			},
+		},
+		{
+			msg: "upstream_bundle is configured correctly",
+			input: func(c *config) {
+				c.Server.UpstreamBundle = true
+			},
+			test: func(t *testing.T, c *server.Config) {
+				require.True(t, c.UpstreamBundle)
+			},
+		},
+		{
+			msg: "allow_agentless_node_attestors is configured correctly",
+			input: func(c *config) {
+				c.Server.Experimental.AllowAgentlessNodeAttestors = true
+			},
+			test: func(t *testing.T, c *server.Config) {
+				require.True(t, c.Experimental.AllowAgentlessNodeAttestors)
+			},
+		},
+		{
+			msg: "bundle endpoint is parsed and configured correctly",
+			input: func(c *config) {
+				c.Server.Experimental.BundleEndpointEnabled = true
+				c.Server.Experimental.BundleEndpointAddress = "192.168.1.1"
+				c.Server.Experimental.BundleEndpointPort = 1337
+			},
+			test: func(t *testing.T, c *server.Config) {
+				require.True(t, c.Experimental.BundleEndpointEnabled)
+				require.Equal(t, "192.168.1.1", c.Experimental.BundleEndpointAddress.IP.String())
+				require.Equal(t, 1337, c.Experimental.BundleEndpointAddress.Port)
+			},
+		},
+		{
+			msg: "svid_ttl is correctly parsed",
+			input: func(c *config) {
+				c.Server.SVIDTTL = "1m"
+			},
+			test: func(t *testing.T, c *server.Config) {
+				require.Equal(t, time.Minute, c.SVIDTTL)
+			},
+		},
+		{
+			msg:         "invalid svid_ttl returns an error",
+			expectError: true,
+			input: func(c *config) {
+				c.Server.SVIDTTL = "b"
+			},
+			test: func(t *testing.T, c *server.Config) {
+				require.Nil(t, c)
+			},
+		},
+		{
+			msg: "ca_ttl is correctly parsed",
+			input: func(c *config) {
+				c.Server.CATTL = "1h"
+			},
+			test: func(t *testing.T, c *server.Config) {
+				require.Equal(t, time.Hour, c.CATTL)
+			},
+		},
+		{
+			msg:         "invalid ca_ttl returns an error",
+			expectError: true,
+			input: func(c *config) {
+				c.Server.CATTL = "b"
+			},
+			test: func(t *testing.T, c *server.Config) {
+				require.Nil(t, c)
+			},
+		},
+		{
+			msg: "ca_subject is configured correctly",
+			input: func(c *config) {
+				c.Server.CASubject = &caSubjectConfig{
+					Organization: []string{"foo"},
+					Country:      []string{"us"},
+					CommonName:   "bar",
+				}
+			},
+			test: func(t *testing.T, c *server.Config) {
+				require.Equal(t, []string{"foo"}, c.CASubject.Organization)
+				require.Equal(t, []string{"us"}, c.CASubject.Country)
+				require.Equal(t, "bar", c.CASubject.CommonName)
+			},
+		},
+	}
+
+	for _, testCase := range cases {
+		input := defaultValidConfig()
+
+		testCase.input(input)
+
+		t.Run(testCase.msg, func(t *testing.T) {
+			sc, err := newServerConfig(input)
+			if testCase.expectError {
+				require.Error(t, err)
```

Comment:
nit: in general i tend to prefer `require.EqualError` over `require.Error` as it asserts that its failing in the way I expect

Reply:
Yes... This had crossed my mind too, however CLI doesn't currently have any error classes defined and I was trying to keep this change scoped to just the fix and the test cases. I'd prefer to not match at all over matching on error string :-| if it's alright with you, I'll punt on error management for now. I can circle back around later and try to improve it - there's a lot of work we can do in the CLI utilities